src/main/dynamo/Beyonce.ts

```diff
@@ -287,6 +277,42 @@ export class Beyonce {
     })
   }
 
+  private async doSingleBatchGet<T extends PartitionAndSortKey<TaggedModel>>(params: {
+    keys: T[]
+    consistentRead?: boolean
+  }): Promise<ExtractKeyType<T>[]> {
+    const useConsistentRead = params.consistentRead !== undefined ? params.consistentRead : this.consistentReads
+    const { Responses: responses, UnprocessedKeys: unprocessedKeys } = await this.client
+      .batchGet({
+        RequestItems: {
+          [this.table.tableName]: {
+            ConsistentRead: useConsistentRead,
+            Keys: params.keys.map(({ partitionKey, sortKey }) => ({
+              [this.table.partitionKeyName]: partitionKey,
+              [this.table.sortKeyName]: sortKey
+            }))
+          }
+        }
+      })
+      .promise()
+
+    if (unprocessedKeys !== undefined && Object.keys(unprocessedKeys).length > 0) {
+      throw new Error(`Some keys didn't process: ${JSON.stringify(unprocessedKeys)}`)
```

Comment:
Can we throw this only if something like a `raiseOnPartialFail` flag is set or something? It seems pretty unforgiving to fail the entire batch if even one bad apple is found. 

For instance, if I'm looking at a list of notes, and one of them for whatever reason is unprocessed, I may still want to display the majority of the notes. As it is now, that one bad apple will break the UI completely.

Reply:
Yea, 100%. I had this as "future work" in the PR description. But, we could spend a bit more time to just do it now. 

Our `batchWrite` method does something similar: 

```TypeScript
  async batchWrite<T extends TaggedModel>(params: {
    putItems?: T[]
    deleteItems?: PartitionAndSortKey<T>[]
  }): Promise<{
    unprocessedPuts: T[]
    unprocessedDeletes: PartitionAndSortKey<T>[]
  }> {
```

And our `batchGet` currently looks like this: 

```TypeScript
  async batchGet<T extends PartitionAndSortKey<TaggedModel>>(params: {
    keys: T[]
    consistentRead?: boolean
  }): Promise<GroupedModels<ExtractKeyType<T>>> {
```

So I think we'd want something like this: 

```TypeScript
  async batchGet<T extends PartitionAndSortKey<TaggedModel>>(params: {
    keys: T[]
    consistentRead?: boolean
  }): Promise<{ 
     items: GroupedModels<ExtractKeyType<T>>,
     unprocessedKeys: PartitionAndSortKey<T>[]
  }> {
```

Reply:
@vly-ginger  -- this is done in https://github.com/ginger-io/beyonce/pull/60/commits/651a89785f892aa15862ea96b003a890ee255f84 part of https://github.com/ginger-io/beyonce/pull/60/